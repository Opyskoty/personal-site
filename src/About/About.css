.About {
  margin-top: 70px !important;
  text-align: center;
  align-items: center;
  padding-bottom: 10px;
  padding-top: 10px;
  font-family: "Inconsolata", monospace;
  font-size: 1.2em;
  border: 1.5px solid rgb(163, 245, 243)!important;
}

.linkedIn {
  height: 300px;
  width: 210px;
  border-radius: 50%;
}

a:hover {
  text-decoration: none !important;
}

a {
  color: rgb(66, 183, 255)!important
}

.skills {
  background-color: rgb(163, 245, 243);
}

h4 {
  margin-bottom: 30px !important;
  margin-top: 15px !important;
}

.microblog {
  width: 180px;
  height: 150px;
  border-radius: 30px;
  margin-right: 30px;
}

.warbler {
  width: 180px;
  height: 150px;
  border-radius: 30px;
  margin-right: 30px;
}
