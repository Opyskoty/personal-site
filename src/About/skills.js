const skills = [
  "javascript",
  "react",
  "node",
  "redux",
  "express",
  "python",
  "flask",
  "sql",
  "html5",
  "css",
  "postgresql",
  "jquery",
];

export default skills;
