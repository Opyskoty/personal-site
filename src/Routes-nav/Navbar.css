.Navbar {
  float: right;
  font-size: 0.9em;
  font-family: "Inconsolata", monospace;
}

a:hover {
  color: lime !important;
}
