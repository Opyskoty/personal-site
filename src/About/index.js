import React from "react";
import { useHistory } from "react-router-dom";
import "./About.css";
import {
  Container,
  Row,
  Col,
  UncontrolledPopover,
  PopoverHeader,
  PopoverBody,
} from "reactstrap";
import { Link } from "react-router-dom";
import linkedImage from "./LinkedIn.jpg";
import microblog from "./Microblog.png";
import color from "./Color.png";
import warbler from "./warbler.png";
import Reward from "react-rewards";
import skills from "./skills";

function About() {
  let reward;
  let history = useHistory();

  const handleClick = (url) => {
    reward.rewardMe();
    setTimeout(function () {
      history.push(url);
    }, 1200);
  };

  return (
    <Container className="About">
      <Row>
        <Col
          md="8"
          className="flexBox"
          style={{
            display: "flex",
            flexFlow: "row wrap",
            justifyContent: "center",
          }}
        >
          <h3>About Me</h3>
          <p>
            I am a full stack software engineer based in SF with a passion for
            React and minimalist designs. I have spent the past few years
            working as an Operations specialist in the tech world, but after
            writing a few sql queries I realized that coding is awesome! This
            led me to self-teach myself some Javascript and eventually complete
            Rithm School's full-stack web development program.
          </p>
          <div>
            When I'm not coding, you can find me hanging out with my
            <Reward
              ref={(ref) => {
                reward = ref;
              }}
              type="confetti"
              config={{ springAnimation: false }}
            >
              <Link to="#" onClick={() => handleClick("/georgie")}>
                dog
              </Link>
              , attempting to cook, or hiking/running around Northern
              California.
            </Reward>
          </div>
          <h4>Things I've Built:</h4>
          <div className="projects">
          <div className="font-italic click">
              click on the icons to find out more
            </div>
            <img
              src={microblog}
              alt="microblog"
              className="microblog"
              type="button"
              id="PopoverLegacy1"
            ></img>
            <UncontrolledPopover
            key="1"
              placement="top"
              trigger="legacy"
              target="PopoverLegacy1"
            >
              <PopoverHeader>MicroBlog</PopoverHeader>
              <PopoverBody>
                A blogging site where users can create, delete, and edit posts,
                as well as vote and comment on posts. Built using React, Redux,
                and PostgreSQL{" "}
                <a
                  href="https://microblog-op.me/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Demo
                </a>{" "}
                |{" "}
                <a
                  href="https://github.com/Opyskoty/microblog"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  GitHub
                </a>
              </PopoverBody>
            </UncontrolledPopover>
            <img
              src={warbler}
              alt="warbler"
              className="warbler"
              type="button"
              id="PopoverLegacy2"
            ></img>
            <UncontrolledPopover
              placement="top"
              trigger="legacy"
              target="PopoverLegacy2"
            >
              <PopoverHeader>Warbler</PopoverHeader>
              <PopoverBody>
                A Twitter clone built using Python, Flask, and PostgreSQL
                <a
                  href="https://github.com/Opyskoty/warbler"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {" "}
                  GitHub
                </a>
              </PopoverBody>
            </UncontrolledPopover>
            <img
              src={color}
              alt="color"
              className="color"
              type="button"
              id="PopoverLegacy3"
            ></img>
            <UncontrolledPopover
              placement="top"
              trigger="legacy"
              target="PopoverLegacy3"
            >
              <PopoverHeader>RGB Color Game</PopoverHeader>
              <PopoverBody>
                Mini game that generates a random RGB code. Objective of the
                game is to find the square with the background color that
                matches the code! Built using jQuery, Bootstrap, HTML and CSS
                <a
                  href="https://opyskoty.github.io/color-game/"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  {" "}
                  Demo
                </a>{" "}
                |{" "}
                <a
                  href="https://github.com/Opyskoty/color-game"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  GitHub
                </a>
              </PopoverBody>
            </UncontrolledPopover>
          </div>
        </Col>
        <Col
          md="4"
          className="flexBox"
          style={{
            display: "flex",
            flexFlow: "row wrap",
            justifyContent: "center",
          }}
        >
          <img className="linkedIn" src={linkedImage} alt="Olivia" />
          <div>
            {skills.map((s) => (
              <React.Fragment key={s.id}>
                <span className="skills">{s.skill}</span>{" "}
              </React.Fragment>
            ))}
          </div>
        </Col>
      </Row>
    </Container>
  );
}
export default About;
