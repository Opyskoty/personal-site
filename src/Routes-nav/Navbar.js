import React from "react";
import { useHistory } from "react-router-dom";
import { Nav, NavItem, NavLink } from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faPaw } from "@fortawesome/free-solid-svg-icons";
import "./Navbar.css";
import Reward from "react-rewards";

function Navbar() {
  const history = useHistory();
  let reward;
  function handleClick(url) {
    reward.rewardMe();
    setTimeout(function () {
      history.push(url);
    }, 1000);
  }
  return (
    <div className="Navbar">
      <Nav>
        <Reward
          ref={(ref) => {
            reward = ref;
          }}
          type="confetti"
        >
          <NavItem>
            <NavLink href="#" onClick={() => handleClick("/")}>home</NavLink>
          </NavItem>
        </Reward>
        <Reward
          ref={(ref) => {
            reward = ref;
          }}
          type="confetti"
        >
          <NavItem>
            <NavLink href="#" onClick={() => handleClick("/about")}>about</NavLink>
          </NavItem>
        </Reward>
        <Reward
          ref={(ref) => {
            reward = ref;
          }}
          type="confetti"
          config={{
            springAnimation: false
          }}
        >
          <NavItem>
            <NavLink href="#" onClick={() => handleClick("/georgie")}>
              <FontAwesomeIcon icon={faPaw}></FontAwesomeIcon>
            </NavLink>
          </NavItem>
        </Reward>
      </Nav>
    </div>
  );
}

export default Navbar;
