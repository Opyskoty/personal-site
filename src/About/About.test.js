import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent } from "@testing-library/react";
import About from "./index";
import Navbar from "../Routes-nav/Navbar";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <About />
    </MemoryRouter>
  );
});

it("handles showing contact modal", async function () {
  const { getByText } = render(
    <MemoryRouter>
      <Navbar>
        <About />
      </Navbar>
    </MemoryRouter>
  );

  fireEvent.click(getByText("contact"));
  expect(getByText("Resume")).toBeInTheDocument();
});

//snapshot
it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <About />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
