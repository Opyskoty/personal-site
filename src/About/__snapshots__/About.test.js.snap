// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="About container"
  >
    <div
      class="row"
    >
      <div
        class="flexBox col-md-8"
        style="display: flex; flex-flow: row wrap; justify-content: center;"
      >
        <h3>
          About Me
        </h3>
        <p>
          I am a full stack software engineer based in SF with a passion for React and minimalist designs. I have spent the past few years working as an Operations specialist in the tech world, but after writing a few sql queries I realized that coding is awesome! This led me to self-teach myself some Javascript and eventually complete Rithm School's full-stack web development program.
        </p>
        <div>
          When I'm not coding, you can find me hanging out with my
          <div />
          <div
            style="position: relative;"
          />
          <div
            style="transform: none;"
          >
            <a
              href="/"
            >
              dog
            </a>
            , attempting to cook, or hiking/running around Northern California.
          </div>
        </div>
        <h4>
          Things I've Built:
        </h4>
        <div
          class="projects"
        >
          <div
            class="font-italic click"
          >
            click on the icons to find out more
          </div>
          <img
            alt="microblog"
            class="microblog"
            id="PopoverLegacy1"
            src="Microblog.png"
            type="button"
          />
          <img
            alt="warbler"
            class="warbler"
            id="PopoverLegacy2"
            src="warbler.png"
            type="button"
          />
          <img
            alt="color"
            class="color"
            id="PopoverLegacy3"
            src="Color.png"
            type="button"
          />
        </div>
      </div>
      <div
        class="flexBox col-md-4"
        style="display: flex; flex-flow: row wrap; justify-content: center;"
      >
        <img
          alt="Olivia"
          class="linkedIn"
          src="LinkedIn.jpg"
        />
        <div>
          <span
            class="skills"
          >
            javascript
          </span>
           
          <span
            class="skills"
          >
            react
          </span>
           
          <span
            class="skills"
          >
            node
          </span>
           
          <span
            class="skills"
          >
            redux
          </span>
           
          <span
            class="skills"
          >
            express
          </span>
           
          <span
            class="skills"
          >
            python
          </span>
           
          <span
            class="skills"
          >
            flask
          </span>
           
          <span
            class="skills"
          >
            sql
          </span>
           
          <span
            class="skills"
          >
            html5
          </span>
           
          <span
            class="skills"
          >
            css
          </span>
           
          <span
            class="skills"
          >
            postgresql
          </span>
           
          <span
            class="skills"
          >
            jquery
          </span>
           
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
